{"name": "test_xfail_not_run", "status": "skipped", "statusDetails": {"message": "_pytest.outcomes.XFailed: [NOTRUN]", "trace": "cls = <class '_pytest.runner.CallInfo'>\nfunc = <function call_runtest_hook.<locals>.<lambda> at 0x10358bdd0>\nwhen = 'setup'\nreraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)\n\n    @classmethod\n    def from_call(cls, func, when, reraise=None) -> \"CallInfo\":\n        #: context of invocation: one of \"setup\", \"call\",\n        #: \"teardown\", \"memocollect\"\n        start = time()\n        excinfo = None\n        try:\n>           result = func()\n\n/usr/local/lib/python3.7/site-packages/_pytest/runner.py:235: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/usr/local/lib/python3.7/site-packages/_pytest/runner.py:208: in <lambda>\n    lambda: ihook(item=item, **kwds), when=when, reraise=reraise\n/usr/local/lib/python3.7/site-packages/pluggy/hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n/usr/local/lib/python3.7/site-packages/pluggy/manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n/usr/local/lib/python3.7/site-packages/pluggy/manager.py:87: in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n/usr/local/lib/python3.7/site-packages/_pytest/skipping.py:87: in pytest_runtest_setup\n    check_xfail_no_run(item)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nitem = <Function test_xfail_not_run>\n\n    def check_xfail_no_run(item):\n        \"\"\"check xfail(run=False)\"\"\"\n        if not item.config.option.runxfail:\n            evalxfail = item._evalxfail\n            if evalxfail.istrue():\n                if not evalxfail.get(\"run\", True):\n>                   xfail(\"[NOTRUN] \" + evalxfail.getexplanation())\nE                   _pytest.outcomes.XFailed: [NOTRUN]\n\n/usr/local/lib/python3.7/site-packages/_pytest/skipping.py:105: XFailed"}, "description": "\n        run=False，表示这条用例不执行\n        :return:\n        ", "start": 1591176793943, "stop": 1591176793943, "uuid": "06dc770b-8d16-46e1-b4f4-448cfa7c1515", "historyId": "e4c8f58bb9381dbcdd35cf7a17339a5f", "testCaseId": "a540d15f54461582abcd1bb98563314d", "fullName": "tests.test_marks.TestMarks#test_xfail_not_run", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_marks"}, {"name": "subSuite", "value": "TestMarks"}, {"name": "host", "value": "LIUMENGJIAdeMacBook-Pro.local"}, {"name": "thread", "value": "78122-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_marks"}]}